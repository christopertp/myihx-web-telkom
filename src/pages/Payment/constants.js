export const BANK_FETCHED = 'Payment/bank-fetched';
export const COPIED = 'Payment/copied';
export const FAILED = 'Payment/failed';
export const FETCHED = 'Payment/fetched';
export const LOADING = 'Payment/loading';
export const STATUS_FETCHED = 'Payment/status-fetched';
