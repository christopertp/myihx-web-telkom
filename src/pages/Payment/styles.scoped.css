.header {
  margin-bottom: 3rem;
}

.header > h2 {
  font-size: 1.25rem;
}

.header > h3 {
  color: #66686a;
  font-size: 0.75rem;
  font-weight: normal;
  letter-spacing: 0.0625rem;
  margin-bottom: 0.25rem;
  margin-top: 1.25rem;
  text-transform: uppercase;
}

.header > p {
  color: #66686a;
  font-size: 0.75rem;
  line-height: 1.25rem;
}

.header > span {
  display: block;
  height: 2rem;
}

.list {
  align-items: center;
  background: #ffffff;
  display: grid;
  grid-template-columns: auto 1fr auto;
  grid-template-rows: auto auto;
  margin-bottom: 0.5rem;
  padding: 0.75rem 0.5rem;
}

.list:last-child {
  margin-bottom: 0;
}

.list > img {
  grid-row: 1/3;
  height: 3rem;
}

.list > img:last-child {
  height: 1.5rem;
}

.list > small {
  color: #66686a;
  font-size: 0.75rem;
  grid-column: 2;
  grid-row: 2;
  margin-left: 1rem;
}

.list > strong {
  color: #333333;
  font-size: 0.9375rem;
  grid-column: 2;
  grid-row: 1;
  line-height: 1.375rem;
  margin-left: 1rem;
}

.loading > span {
  border-radius: 0.5rem;
  display: block;
  height: 4.5rem;
  margin-bottom: 0.5rem;
}

.timer {
  background-color: #115b79;
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 1, 0.15);
  color: #ffffff;
  border-radius: 1.125rem;
  display: grid;
  grid-template-columns: auto 1fr;
  padding: 0.4rem;
  float: right;
}

.timer > img {
  padding: 0.1rem;
  width: 1.25rem;
}

.timer > span {
  line-height: 1.2rem;
}

@media only screen and (max-width: 48em) {
  .header::after {
    background: url(/assets/bg_header_white_long.svg) no-repeat;
    background-size: cover;
    content: '';
    height: 16.75rem;
    left: 0;
    position: absolute;
    top: 0;
    width: 100%;
    z-index: -1;
  }

  .timer > span {
    font-size: 1rem;
  }
}
