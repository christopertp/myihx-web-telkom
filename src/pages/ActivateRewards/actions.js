import { RESET, LOADING, FAILED, FETCHED, STATUS, USER, COUNTER } from './constants';
import { push } from 'react-router-redux';
import { profile } from '../../services/user';
import { postActivateReward, postVerifyOtp, postResendOtp, statusPoints } from '../../services/rewards';

export function activateReward(category, type) {
  return async dispatch => {
    dispatch(loadingAction(true));
    try {
      const request = {
        'category': category,
        'type': type,
      };
      const { data } = await postActivateReward(request);
      dispatch(timeOTPFetchedAction(data));
    } catch (err) {
      dispatch(failedAction(err.message));
    }
  };
}

export function verifyOtp(payload) {
  return async dispatch => {
    dispatch(loadingAction(true));
    try {
      const request = {
        'otp': payload,
      };
      const { data } = await postVerifyOtp(request);
      if (data.status === 'success') {
        dispatch(push('/activate-rewards/success'));
      } else {
        dispatch(push('/activate-rewards'));
      }
    } catch (err) {
      const { message, data } = err;
      dispatch(failedAction(err.message));
      if (err?.message === 'Otp not match') {
        dispatch(counterAction(data, message));
      }
    }
  };
}

export function fetchStatusRewards() {
  return async dispatch => {
    dispatch(loadingAction(true));
    try {
      const { data } = await statusPoints();
      dispatch(statusAction(data));
    } catch (err) {
      dispatch(failedAction(err.message));
    }
  };
}

export function resendOtp(type) {
  return async dispatch => {
    dispatch(loadingAction(true));
    try {
      const request = {
        'type': type
      };
      const { data } = await postResendOtp(request);
      dispatch(timeOTPFetchedAction(data));
    } catch (err) {
      dispatch(failedAction(err.message));
    }
  };
}

export function fetchDataUser() {
  return async dispatch => {
    dispatch(loadingAction(true));
    try {
      const { data } = await profile();
      dispatch(loadingAction(false));
      dispatch(userAction(data));
    } catch (err) {
      dispatch(failedAction(err.message));
    }
  };
}

function timeOTPFetchedAction(data) {
  const nowDate = new Date().toISOString();
  const diffDate = Date.parse(data.deadlineOTP) - Date.parse(nowDate);
  const timeLeft = parseInt(diffDate / 1000);
  data = {
    ...data,
    deadlineOTP: timeLeft,
    isLoading: false,
  };
  return { type: FETCHED, data };
}

function loadingAction(isLoading) {
  return { type: LOADING, isLoading };
}

function failedAction(message) {
  return { type: FAILED, message };
}

function statusAction(data) {
  return { type: STATUS, data };
}

function userAction(data) {
  return { type: USER, data };
}

function counterAction(data, message) {
  return { type: COUNTER, data, message };
}

export function resetData() {
  return { type: RESET };
}
