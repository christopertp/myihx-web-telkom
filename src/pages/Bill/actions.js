import { billingInfo } from '../../services/billing';
import { FAILED, FETCHED, LOADING } from './constants';

export function fetchBill() {
  return async dispatch => {
    dispatch(loadingAction(true));
    try {
      const { data } = await billingInfo();
      const { billingPeriod, details, dueDate, status, totalAmount } = data;
      dispatch(fetchedAction(billingPeriod, details, dueDate, status, totalAmount));
    } catch(err) {
      dispatch(failedAction(err.message));
    }
  };
}

function failedAction(message) {
  return { type: FAILED, message };
}

function fetchedAction(billingPeriod, details, dueDate, status, totalAmount) {
  return { type: FETCHED, billingPeriod, details, dueDate, status, totalAmount };
}

function loadingAction(isLoading) {
  return { type: LOADING, isLoading };
}
