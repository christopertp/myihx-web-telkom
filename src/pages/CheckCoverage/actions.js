import { push } from 'react-router-redux';
import * as order from '../../services/order';
import { availability, guestAvailability } from '../../services/product';
import { addressCities, addressDistricts, addressProvinces, addressStreets } from '../../services/user';
import { updateSyncErrors } from '../../utils/redux-form-actions';
import { clearGuestAddress, getToken, setGuestAddress } from '../../utils/storage';
import { DATA_FETCHED, FAILED, LOADING } from './constants';

export function fetchAddress(key, id1, id2, id3) {
  const apis = {
    cities: addressCities,
    districts: addressDistricts,
    provinces: addressProvinces,
    streets: addressStreets,
  };
  const fields = {
    cities: 'cityCode',
    districts: 'districtCode',
    provinces: 'provinceCode',
    streets: 'streetCode',
  };
  return async dispatch => {
    dispatch(loadingAction(true, key.charAt(0)));
    try {
      const { data } = await apis[key](id1, id2, id3);
      dispatch(dataFetchedAction(data, key));
    } catch (err) {
      const message = err.code === 404 ? 'Data tidak ditemukan' : 'Gagal memuat data';
      dispatch(dataFetchedAction([], key));
      dispatch(updateSyncErrors('checkCoverageAddress', fields[key], message));
    }
  };
}

export function fetchAvailability(payload) {
  return async dispatch => {
    const fn = getToken() ? availability : guestAvailability;
    dispatch(loadingAction(true, 'submit'));
    try {
      const { data } = await fn(payload);
      if (data.status === 'PT1') {
        setGuestAddress({
          ...data.installationAddress,
          addressDescription: payload.installationAddress.addressDescription,
        });
        dispatch(failedAction(''));
        dispatch(push('/check-coverage/pt1'));
      } else {
        dispatch(failedAction('Status unknown'));
      }
    } catch (err) {
      const { data, message } = err;
      if (!data) {
        dispatch(failedAction(message));
      } else if (data.status === 'PT2') {
        dispatch(failedAction('PT2'));
        clearGuestAddress();
      } else if (data.status === 'PT3') {
        dispatch(failedAction(''));
        dispatch(push('/check-coverage/pt3'));
        clearGuestAddress();
      } else {
        dispatch(failedAction(message));
      }
    }
  };
}

export function fetchSubscribe() {
  return async dispatch => {
    dispatch(loadingAction(true, 'submit'));
    try {
      await order.getSubscriptions();
      dispatch(push('/personal-data'));
    } catch (err) {
      dispatch(push('/shop/internet'));
      dispatch(loadingAction(false, 'submit'));
    }
  };
}

export function resetMessage() {
  return failedAction('');
}

function dataFetchedAction(data, key) {
  return { type: DATA_FETCHED, data, key };
}

function failedAction(message) {
  return { type: FAILED, message };
}

function loadingAction(isLoading, key) {
  return { type: LOADING, isLoading, key };
}
