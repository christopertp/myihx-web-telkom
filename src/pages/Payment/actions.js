import { paymentMethod, paymentBankTransfer, paymentLinkAjaDebitCC, paymentGetStatus } from '../../services/payment';
import { COPIED, FAILED, FETCHED, LOADING, BANK_FETCHED, STATUS_FETCHED } from './constants';
import { push } from 'react-router-redux';

export function fetchMethods(method, transactionIdMethod) {
  return async dispatch => {
    dispatch(loadingAction(true, 'p'));
    try {
      const { data } = await paymentMethod(method, transactionIdMethod);
      const { methods, amount, bankInfo, transactionId, deadline } = data;
      dispatch(fetchedAction(amount, transactionId, bankInfo, methods, deadline));
    } catch (err) {
      dispatch(failedAction(err.message));
    }
  };
}

export function fetchBankTransfer(method, amount, codeBank, code, detailBank) {
  const splitCode = code.split('-');
  return async dispatch => {
    dispatch(loadingAction(true, 'b'));
    try {
      const { data } = await paymentBankTransfer(method, amount, codeBank, code, 'BANK_TRANSFER');
      const { paymentCode, deadline, transactionId } = data;
      dispatch(fetchedBankAction(codeBank, paymentCode, detailBank, deadline, transactionId));
      dispatch(push(`bank-transfer/${codeBank}`));
    } catch (err) {
      dispatch(push(`/payment-${code}/failed?invoice=${splitCode[1]}`));
    }
  };
}

export function fetchLinkAjaDebitCC(payload, method) {
  const { amount, transactionId, type } = method;
  const splitTransactionId = transactionId.split('-');
  return async dispatch => {
    try {
      const { data } = await paymentLinkAjaDebitCC(payload, amount, transactionId, type);
      window.location.href = `${data.redirectUrl}`;
    } catch (err) {
      dispatch(push(`/payment-${transactionId}/failed?invoice=${splitTransactionId[1]}`));
    }
  };
}

export function fetchPaymentStatus(transactionId) {
  return async dispatch => {
    dispatch(loadingAction(true, 's'));
    try {
      const { data } = await paymentGetStatus(transactionId);
      const { productInfo, customer, deadline, paymentType, paymentStatus, totalDue } = data;
      dispatch(fetchedStatusAction(productInfo, customer, deadline,
        paymentType, paymentStatus, totalDue));
    } catch (err) {
      dispatch(failedAction(err.message));
    }
  };
}

export function setMessageAction(message) {
  return { type: COPIED, message };
}

function failedAction(message) {
  return { type: FAILED, message };
}

function fetchedAction(amount, transactionId, bankInfo, methods, deadline) {
  return { type: FETCHED, amount, transactionId, bankInfo, methods, deadline };
}

function loadingAction(loading, key) {
  return { type: LOADING, loading, key };
}

function fetchedBankAction(codeBank, paymentCode, detailBank, deadline, transactionId) {
  return { type: BANK_FETCHED, codeBank, paymentCode, detailBank, deadline, transactionId };
}

function fetchedStatusAction(productInfo, customer, deadline, paymentType,
  paymentStatus, totalDue) {
  return { type: STATUS_FETCHED, productInfo, customer, deadline,
    paymentType, paymentStatus, totalDue };
}
