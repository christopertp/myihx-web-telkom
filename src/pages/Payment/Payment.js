import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useParams, useHistory } from 'react-router-dom';
import Button from '../../components/elements/Button';
import Card from '../../components/elements/Card';
import PaymentBankTransfer from '../../components/fragments/PaymentBankTransfer';
import PaymentSuccess from '../../components/fragments/PaymentSuccess';
import PaymentFailed from '../../components/fragments/PaymentFailed';
import WhiteCardPlain from '../../components/layouts/WhiteCardPlain';
import useTimer from '../../hooks/useTimer';
import { thousand } from '../../utils/format';
import { fetchMethods, fetchLinkAjaDebitCC } from './actions';
import styles from './styles.scoped.css';

export default function Payment() {
  const { method, detail } = useParams();

  if (detail === 'bank-transfer') {
    return <PaymentBankTransfer />;
  }

  return (m => {
    switch (m) {
      case 'deposit' : return <Method />;
      case 'failed': return <PaymentFailed />;
      case 'postpaid' : return <Method />;
      case 'prepaid' : return <Method />;
      case 'success': return <PaymentSuccess />;
      default: return <NotExist />;
    }
  })(method);
}

export function Method() {
  const dispatch = useDispatch();
  const { method, code } = useParams();
  const { amount, transactionId, loading, message } = useSelector(s => s.payment);

  useEffect(() => {
    if (method) {
      dispatch(fetchMethods(method, code));
    }
  }, [method]);

  if (message) {
    return <NotExist />;
  }

  return (
    <WhiteCardPlain size="medium" title="Payment">
      <div className={styles.header}>
        {loading.p ? <span className="loading" /> : (method === 'deposit') && <Timer />}
        <h3>Payment Due</h3>
        {loading.p ? <span className="loading" /> : <h2>Rp{thousand(amount)}</h2>}
        <p>Includes PPn 10%</p>
        <h3>Payment Code</h3>
        {loading.p ? <span className="loading" /> : <h2>{transactionId}</h2>}
      </div>
      {
        loading.p ?
          <div className={styles.loading}>
            <span className="loading" />
            <span className="loading" />
            <span className="loading" />
          </div> :
          <List />
      }
    </WhiteCardPlain>
  );
}

export function List() {
  const dispatch = useDispatch();
  const history = useHistory();
  const { method } = useParams();
  const { amount, transactionId, methods } = useSelector(s => s.payment);

  return methods.map(v => {
    const status = {
      BANK_TRANSFER: { onClick: () => { history.push(`${method}/bank-transfer`); } },
      DEBIT_CC: { onClick: () => { dispatch(fetchLinkAjaDebitCC(method, { amount, transactionId, type: 'DEBIT_CC', })); } },
      LINK_AJA: { onClick: () => { dispatch(fetchLinkAjaDebitCC(method, { amount, transactionId, type: 'LINK_AJA', })); } },
      TMONEY: { onClick: () => {} },
    };

    return (
      <Card className={styles.list} key={v.methodId} onClick={status[v.type].onClick}>
        <img alt={v.label} src={v.icon} />
        <strong>{v.label}</strong>
        <small>{v.info}</small>
        <img alt="chevron_red" src="/assets/ic_chevron_red.svg" />
      </Card>
    );
  });
}

export function NotExist() {
  return (
    <WhiteCardPlain icon="warning" size="medium" title="Payment does not exist">
      <Button to="/">Back to home</Button>
    </WhiteCardPlain>
  );
}

export function Timer() {
  const { deadline } = useSelector(s => s.payment);
  const currentTime = new Date();
  const expireTime = new Date(deadline);
  const timeInSeconds = (expireTime - currentTime) / (1000 * 60) * 60;

  const { time } = useTimer(Math.floor(timeInSeconds));
  const [hour, min] = time;
  return (
    <div className={styles.timer}>
      <img alt="time" src="/assets/ic_timer.svg" />
      <span>{hour}:{min}</span>
    </div>
  );
}
