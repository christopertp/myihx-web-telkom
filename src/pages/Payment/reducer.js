import { fetchBankTransfer, fetchPaymentStatus, setMessageAction } from './actions';
import { BANK_FETCHED, COPIED, FAILED, FETCHED, LOADING, STATUS_FETCHED } from './constants';

const initialState = {
  amount: 0,
  bankInfo: '',
  codeBank: '',
  customer: '',
  deadline: '',
  detailBank: '',
  fetchBankTransfer,
  fetchPaymentStatus,
  loading: { p: true, b: true, s: true },
  message: '',
  methods: '',
  paymentCode: '',
  paymentType: '',
  paymentStatus: '',
  productInfo: '',
  setMessageAction,
  transactionId: '',
  totalDue: 0,
};

export default function reducer(state = initialState, action = {}) {
  const { type, amount, bankInfo, loading, key, message, methods,
    codeBank, paymentCode, detailBank, deadline, transactionId, productInfo,
    customer, paymentType, paymentStatus, totalDue } = action;

  switch (type) {
    case BANK_FETCHED:
      return { ...state, loading: { ...state.loading, b: false }, message:'', codeBank, paymentCode, detailBank, deadline, transactionId };
    case STATUS_FETCHED:
      return { ...state, loading: { ...state.loading, s: false }, message:'', productInfo, customer, deadline, paymentType, paymentStatus, totalDue };
    case COPIED:
      return { ...state, loading: { ...state.loading, b: false }, message };
    case FAILED:
      return { ...state, loading: { ...state.loading, p: false }, message };
    case FETCHED:
      return { ...state, loading: { ...state.loading, p: false }, message: '', amount, transactionId, bankInfo, methods, deadline };
    case LOADING:
      return { ...state, loading: { ...state.loading, [key]: loading } };
    default:
      return state;
  }
}
