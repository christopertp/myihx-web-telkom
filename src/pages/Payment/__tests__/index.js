import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useParams, useHistory } from 'react-router-dom';
import ShallowRenderer from 'react-test-renderer/shallow';
import Payment, { Method, List, NotExist, Timer } from '../Payment';
import { fetchMethods } from '../actions';
import PaymentFailed from '../../../components/fragments/PaymentFailed';
import PaymentSuccess from '../../../components/fragments/PaymentSuccess';
import useTimer from '../../../hooks/useTimer';
import PaymentBankTransfer from '../../../components/fragments/PaymentBankTransfer';

jest.mock('../actions', () => ({
  fetchLinkAjaDebitCC: jest.fn(),
  fetchMethods: jest.fn(),
}));

const methods = [{
  icon: '',
  info: 'Pay with LinkAja',
  label: 'LinkAja',
  type: 'LINK_AJA',
  url: 'http://linkaja.com/1234/21312313',
},
{
  icon: '',
  info: 'Pay with DebitCC',
  label: 'DebitCC',
  type: 'DEBIT_CC',
  url: 'http://debit.com/1234/21312313',
},
{
  icon: '',
  info: 'Pay with TMONEY',
  label: 'TMoney',
  type: 'TMONEY',
  url: 'http://tmoney.com/1234/21312313',
},
{
  icon: '',
  info: 'Pay with Bank Transfer',
  label: 'LinkAja',
  type: 'BANK_TRANSFER',
  url: 'http://linkaja.com/1234/21312313',
},];

useSelector.mockImplementation(fn => {
  fn({});
  return {
    amount: 396525,
    methods,
    loading: { p: false },
    deadline: '2020-06-27T03:00:00.043Z',
    transactionId: 'MYIRX-986401919709',
  };
});

jest.mock('../../../hooks/useTimer', () => jest.fn(() => ({
  time: ['', '', ''],
})));

describe('src/pages/Payment', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Payment />);
    expect(tree).toMatchSnapshot();

    useParams.mockImplementationOnce(() => ({ method: 'failed' }));
    const result = Payment();
    expect(result.type).toBe(PaymentFailed);

    useParams.mockImplementationOnce(() => ({ method: 'deposit' }));
    const result2 = Payment();
    expect(result2.type).toBe(Method);

    useParams.mockImplementationOnce(() => ({ method: 'postpaid' }));
    const result3 = Payment();
    expect(result3.type).toBe(Method);

    useParams.mockImplementationOnce(() => ({ method: 'prepaid' }));
    const result4 = Payment();
    expect(result4.type).toBe(Method);

    useParams.mockImplementationOnce(() => ({ method: 'success' }));
    const result5 = Payment();
    expect(result5.type).toBe(PaymentSuccess);

    useParams.mockImplementationOnce(() => ({ detail: 'bank-transfer' }));
    const result6 = Payment();
    expect(result6.type).toBe(PaymentBankTransfer);
  });

  test('Method', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Method />);
    expect(tree).toMatchSnapshot();
  });

  test('List', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<List />);
    expect(tree).toMatchSnapshot();
  });

  test('List call Bank Transfer', () => {
    const methods =  [{
      icon: '',
      info: 'Pay with Bank',
      label: 'Bank Transfer',
      type: 'BANK_TRANSFER',
      url: 'http://linkaja.com/1234/21312313',
    }];

    const history = { push: jest.fn() };
    useHistory.mockImplementationOnce(() => history);
    useSelector.mockImplementationOnce(() => ({ methods }));
    const result = List();
    result[0].props.onClick();
    expect(history.push).toHaveBeenCalled();
  });

  test('List call Debit CC', () => {
    const methods =  [{
      icon: '',
      info: 'Pay with DebitCC',
      label: 'DebitCC',
      type: 'DEBIT_CC',
      url: 'http://debit.com/1234/21312313',
    }];

    const dispatch = jest.fn();
    useDispatch.mockImplementationOnce(() => dispatch);
    useSelector.mockImplementationOnce(() => ({ methods }));
    const result = List();
    result[0].props.onClick();
    expect(dispatch).toHaveBeenCalled();
  });

  test('List call LinkAja', () => {
    const methods =  [{
      icon: '',
      info: 'Pay with LinkAja',
      label: 'LinkAja',
      type: 'LINK_AJA',
      url: 'http://linkaja.com/1234/21312313',
    }];

    const dispatch = jest.fn();
    useDispatch.mockImplementationOnce(() => dispatch);
    useSelector.mockImplementationOnce(() => ({ methods }));
    const result = List();
    result[0].props.onClick();
    expect(dispatch).toHaveBeenCalled();
  });

  test('List call TMONEY', () => {
    const methods =  [{
      icon: '',
      info: 'Pay with TMoney',
      label: 'TMoney',
      type: 'TMONEY',
      url: 'http://tmoney.com/1234/21312313',
    }];

    useSelector.mockImplementationOnce(() => ({ methods }));

    const result = List();
    result[0].props.onClick();
  });

  test('NotExist', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<NotExist />);
    expect(tree).toMatchSnapshot();
  });

  test('render loading state', () => {
    useSelector.mockImplementationOnce(() => ({ loading: { p: true } }));

    const result = Method();
    expect(result.props.children[0].props.children[2].props.className).toBe('loading');
    expect(result.props.children[1].props.children[0].props.className).toBe('loading');
  });

  test('render not exist', () => {
    useSelector.mockImplementationOnce(
      () => ({ message: '"type" must be one of [postpaid, prepaid, deposit]' })
    );

    const result = Method();
    expect(result.type).toBe(NotExist);
  });

  test('call get payment methods', () => {
    const dispatch = jest.fn();
    useDispatch.mockImplementationOnce(() => dispatch);
    useParams.mockImplementationOnce(() => ({ method: 'deposit', code: 'MYIRX-986401919709' }));

    Method();
    expect(dispatch).toHaveBeenCalled();
    expect(fetchMethods).toHaveBeenCalledWith('deposit', 'MYIRX-986401919709');
  });

  test('Timer', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Timer />);
    expect(tree).toMatchSnapshot();
    useTimer.mockImplementationOnce(() => ({
      time: ['', '', ''],
    }));
  });
});
