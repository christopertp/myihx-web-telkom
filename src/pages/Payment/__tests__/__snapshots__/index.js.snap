// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/pages/Payment List 1`] = `
Array [
  <Card
    className=""
    disabled={false}
    onClick={[Function]}
    to=""
    variant="bordered"
  >
    <img
      alt="LinkAja"
      src=""
    />
    <strong>
      LinkAja
    </strong>
    <small>
      Pay with LinkAja
    </small>
    <img
      alt="chevron_red"
      src="/assets/ic_chevron_red.svg"
    />
  </Card>,
  <Card
    className=""
    disabled={false}
    onClick={[Function]}
    to=""
    variant="bordered"
  >
    <img
      alt="DebitCC"
      src=""
    />
    <strong>
      DebitCC
    </strong>
    <small>
      Pay with DebitCC
    </small>
    <img
      alt="chevron_red"
      src="/assets/ic_chevron_red.svg"
    />
  </Card>,
  <Card
    className=""
    disabled={false}
    onClick={[Function]}
    to=""
    variant="bordered"
  >
    <img
      alt="TMoney"
      src=""
    />
    <strong>
      TMoney
    </strong>
    <small>
      Pay with TMONEY
    </small>
    <img
      alt="chevron_red"
      src="/assets/ic_chevron_red.svg"
    />
  </Card>,
  <Card
    className=""
    disabled={false}
    onClick={[Function]}
    to=""
    variant="bordered"
  >
    <img
      alt="LinkAja"
      src=""
    />
    <strong>
      LinkAja
    </strong>
    <small>
      Pay with Bank Transfer
    </small>
    <img
      alt="chevron_red"
      src="/assets/ic_chevron_red.svg"
    />
  </Card>,
]
`;

exports[`src/pages/Payment Method 1`] = `
<WhiteCardPlain
  back=""
  close=""
  help={false}
  icon=""
  size="medium"
  subtitle=""
  title="Payment"
>
  <div>
    <h3>
      Payment Due
    </h3>
    <h2>
      Rp
      396.525
    </h2>
    <p>
      Includes PPn 10%
    </p>
    <h3>
      Payment Code
    </h3>
    <h2>
      MYIRX-986401919709
    </h2>
  </div>
  <List />
</WhiteCardPlain>
`;

exports[`src/pages/Payment NotExist 1`] = `
<WhiteCardPlain
  back=""
  close=""
  help={false}
  icon="warning"
  size="medium"
  subtitle=""
  title="Payment does not exist"
>
  <Button
    className=""
    disabled={false}
    onClick={null}
    to="/"
    type="button"
    variant="red"
  >
    Back to home
  </Button>
</WhiteCardPlain>
`;

exports[`src/pages/Payment Timer 1`] = `
<div>
  <img
    alt="time"
    src="/assets/ic_timer.svg"
  />
  <span>
    
    :
    
  </span>
</div>
`;

exports[`src/pages/Payment render 1`] = `<NotExist />`;
